dwave/optimization/mathematical.py

```diff
@@ -1237,6 +1239,39 @@ def safe_divide(x1: ArraySymbol, x2: ArraySymbol) -> SafeDivide:
     return SafeDivide(x1, x2)
 
 
+def softmax(array: ArraySymbol) -> SoftMax:
+    """Return softmax of a given symbol. Given a flattened array 
+    x: [x_1, x_2, ..., x_n], softmax(x) returns an array [y_1, y_2, ..., y_n] 
+    such that y_i = exp(x_i) / (exp(x_1) + exp(x_2) + ... + exp(x_n)).
+
+    Args:
+        array: Input array symbol.
+
+    Example:
+        This example computes the softmax of one symbol.
+
+        >>> from dwave.optimization import Model
+        >>> from dwave.optimization.mathematical import softmax
+        >>> import numpy as np
+        ...
+        >>> model = Model()
+        >>> i = model.integer(3)
+        >>> sm = softmax(i)
+        >>> expected = np.array([0.0900305731703, 0.6652409557748, 0.244728471054])
+        >>> with model.lock():
+        ...     model.states.resize(1)
+        ...     i.set_state(0, [1, 3, 2])
+        ...     print(sum(np.isclose(sm.state(0), expected)))
```

Comment:
Minor nit but I might frame this as `print(np.isclose(sm.state(), expected).all())`